cukinia

```diff
@@ -565,6 +566,27 @@ _cukinia_process()
 	return 1
 }
 
+# cukinia_kprocess: checks if kernel process $pname runs
+# arg: $1: the kernel process name
+_cukinia_kprocess()
+{
+	local pname="$1"
+	local procdir
+	local pid
+
+	_cukinia_prepare "Checking kernel process \"$pname\" ${__not:+NOT }running"
+
+	# scan /proc to get process name
+        # as some version of pidof do not look for kernel process, use pgrep with exact match
+	for pid in $(pgrep -x "$pname"); do
+		# check that it is a kernel threads
+                # null byte is removed from thread cmdline to avoid bash warning
+		[ -z "$(tr -d '\0' < /proc/$pid/cmdline)" ] && return 0
```

Comment:
Some userspace processes rewrite their cmdline which could result in passing this test. What about testing that `readlink proc/$pid/exe` is empty, a userspace process will always be started from a non-null executable

Reply:
I'm not sure which is better. I just tried with readlink and systemd looks has a bit of a weird behavior:

- its cmdline in /sbin/initsplash and detected as a user process
- its exe is empty and detected as a kernel thread

`ps` looks like it detects it as a user process too. Which one is correct ?

Reply:
This is not ideal as non-determinstic. Maybe a few other differentiators in /proc/${pid}, maps will be empty for kernel threads, but shall not be for user processes (whether static or dynamic).

Reply:
"maps" is only readable by root or in sudo, so would break some tests.
I'll be looking at alternatives and update this PR when I find something deterministic and correct.

Reply:
The more accurate alternative that seems the easiest to test from cukinia is the presence of a VmSize attribute in the status file. Kernel threads do not have a VMSize obviously. Pushing a commit for that.